demo/Dockerfile

```diff
@@ -26,14 +26,14 @@ FROM jenkinsci/workflow-demo:2.2
 
 USER root
 
-RUN apt-get update
-RUN apt-get install -y sudo
-RUN adduser jenkins sudo
-RUN echo 'Defaults !authenticate' >> /etc/sudoers
+RUN apt-get update && \
+    apt-get install -y sudo
+RUN adduser jenkins sudo && \
+    echo 'Defaults !authenticate' >> /etc/sudoers
 
 # Minimum for docker-workflow. Docker server does not accept a client newer than itself (older is OK).
-RUN wget -O /usr/bin/docker --no-check-certificate https://get.docker.com/builds/Linux/x86_64/docker-1.4.0
-RUN chmod a+x /usr/bin/docker
+RUN wget --no-verbose -O /usr/bin/docker --no-check-certificate https://get.docker.com/builds/Linux/x86_64/docker-1.7.0 && \
+    chmod a+x /usr/bin/docker
```

Comment:
This change was for three reasons:
1) Upgrade the docker client to v1.7.0 (the important change)
2) To get rid of the scrolling wall of text that occurred due to the output expecting to be interactive.  It was printing a statusbar to the log on a given interval
3) Prevent docker from caching unnecessary additional layers.  The Dockerfile best practices recommend concatenating your run commands to prevent caching of some layers.  I stopped short of doing it to the entire file and settled for the two related items.  See: [Dockerfile best practices](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/)